test-support/test_oidc_provider.js

```diff
@@ -7,30 +7,32 @@ const configuration = {
       response_types: ['id_token'],
       response_mode: ['form_post'],
       grant_types: ['implicit'],
-      // "native" because we're on localhost.
-      application_type: 'native',
+      application_type: 'web',
       scopes: ['openid', 'profile'],
       redirect_uris: [
         'http://localhost:9000/callback/OidcClient',
         'http://localhost:9000/callback/AdClient',
         'http://localhost:19001/callback/OidcClient',
         'http://localhost:19001/callback/AdClient',
+        'http://civiform:9000/callback/OidcClient',
```

Comment:
Likewise can you suss out where localhost and civiform are separately relevant?  This file in generally is hard to follow how we use it.

Pointing people to the invalidate below or making a file level comment for both works.

Reply:
Attempted to clarify these. I *think* there's an opportunity to get rid of the 19001 callbacks entirely, but I'll do that separately after this has been submitted.